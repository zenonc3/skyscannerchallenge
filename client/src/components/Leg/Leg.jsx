import BpkText from 'bpk-component-text/src/BpkText';
import PropTypes from 'prop-types';
import React from 'react';
import STYLES from './Leg.scss';
import TimeLocation from '../TimeLocation';
import LegMeta from '../LegMeta';

const c = className => STYLES[className] || 'UNKNOWN';

const Leg = ({
  origin, destination, departure, arrival, duration, numStops, carrier,
}) => (
  <div className={c('Leg__container')} >
    <span><img src={`https://logos.skyscnr.com/images/airlines/favicon/${carrier}.png`} className={c('Leg__logo')} alt="airline logo" /></span>
    <div className={c('Leg__origin')}><TimeLocation time={departure} location={origin} /></div>
    <div className={c('Leg__arrow')}><BpkText>â†’</BpkText></div>
    <div className={c('Leg__destination')}><TimeLocation time={arrival} location={destination} /></div>
    <div className={c('Leg__meta')}><LegMeta numStops={numStops} duration={duration} /></div>
  </div>
);

Leg.propTypes = {
  origin: PropTypes.string.isRequired,
  destination: PropTypes.string.isRequired,
  departure: PropTypes.string.isRequired,
  arrival: PropTypes.string.isRequired,
  duration: PropTypes.number.isRequired,
  numStops: PropTypes.number.isRequired,
  carrier: PropTypes.string.isRequired,
};

export default Leg;
