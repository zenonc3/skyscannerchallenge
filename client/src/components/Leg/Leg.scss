@import '~bpk-mixins';

.Leg {
  &__container {
    display: flex;
    width: 100%;
    padding: $bpk-spacing-xs;
    flex-direction: row;
  }

  &__logo {
    display: flex;
    height: 35pt;
    flex-direction: column;
    flex-basis: 35pt;
    border-radius: $bpk-border-radius-sm;
  }

  &__origin {
    display: flex;
    height: 35pt;
    margin-left: $bpk-spacing-xs;
    flex-direction: column;
    align-content: center;
    flex-basis: 35pt;
  }

  &__destination {
    display: flex;
    height: 35pt;
    margin-left: $bpk-spacing-xs;
    flex-direction: column;
    align-content: center;
    flex-basis: 35pt;
  }

  &__meta {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-end;
    flex-grow: 1;
  }

  &__arrow {
    display: flex;
    height: 35pt;
    align-items: center;
  }
}
