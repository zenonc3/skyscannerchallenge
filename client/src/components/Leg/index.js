import Leg from './Leg';

export default Leg;
