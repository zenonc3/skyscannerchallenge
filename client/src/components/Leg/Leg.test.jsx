import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Leg from './Leg';


describe('Itinerary', () => {
  const dummyLeg = {
    origin: '',
    destination: '',
    departure: '',
    arrival: '',
    duration: 10,
    numStops: 1,
    carrier: '',
  };
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Leg {...dummyLeg} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

