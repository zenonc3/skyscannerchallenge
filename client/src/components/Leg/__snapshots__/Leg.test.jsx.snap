// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary should render correctly 1`] = `
<div
  className="Leg__container"
>
  <span>
    <img
      alt="airline logo"
      className="Leg__logo"
      src="https://logos.skyscnr.com/images/airlines/favicon/.png"
    />
  </span>
  <div
    className="Leg__origin"
  >
    <TimeLocation
      location=""
      time=""
    />
  </div>
  <div
    className="Leg__arrow"
  >
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="base"
    >
      â†’
    </BpkText>
  </div>
  <div
    className="Leg__destination"
  >
    <TimeLocation
      location=""
      time=""
    />
  </div>
  <div
    className="Leg__meta"
  >
    <LegMeta
      duration={10}
      numStops={1}
    />
  </div>
</div>
`;
